packages/workspace/src/browser/workspace-service.ts

```diff
@@ -66,7 +66,9 @@ export class WorkspaceService implements FrontendApplicationContribution {
     protected readonly schemaProvider: PreferenceSchemaProvider;
 
     protected applicationName: string;
-
+    protected readonly onDidCreateFileFolderEmitter = new Emitter<string>();
```

Comment:
I'm not sure that the workspace service should have such APIs.

So if someone call create a folder programatically, this file will be always revealed in the navigator? Is it desirable? I think the scope of the change should be bound to the navigator.

Reply:
There are already APIs btw to listen when something is created: `FileSystemWatcher`. And it should be used.

Reply:
But we don't want to highlight any folder that gets created. Right now creating a folder in the navigator tree triggers a command from the workspace extension. How would you fix https://github.com/eclipse-theia/theia/issues/6190 ?

Reply:
> But we don't want to highlight any folder that gets created.

Changing WorkspaceService does not help, anyone can trigger programmatically these methods now and it would lead to the same issues.

>  How would you fix #6190 ?

If it should be triggered only by user actions then it should be implemented there, like await then create finished (without listening to events), if a result is positive then reveal. `WorkspaceService` should not be changed.